pkgs/by-name/st/sta/package.nix

```diff
@@ -16,8 +17,28 @@ stdenv.mkDerivation {
     sha256 = "sha256-AiygCfBze7J1Emy6mc27Dim34eLR7VId9wodUZapIL4=";
   };
 
+  strictDeps = true;
+
   nativeBuildInputs = [ autoreconfHook ];
 
+  doCheck = true;
+  nativeCheckInputs = [ cxxtest ];
+  checkInputs = [ cxxtest ];
+
+  checkPhase = ''
+    runHook preCheck
+
+    pushd test
+
+    cxxtestgen --error-printer --have-std -o tests.cpp sta_test_1.h sta_test_2.h
+    ${stdenv.cc.targetPrefix}c++ -o tester tests.cpp
+    ./tester
+
+    popd
+
+    runHook postCheck
+  '';
+
   meta = with lib; {
```

Comment:
Consider removing with lib; as it increases eval time while we're at all this. Certainly not required.